### ISSUE TITLE
FR: Update camera APIs in the MLKit samples and improve performance

### ISSUE THREAD

**Issue by samtstern** [firebase/quickstart-android#842]
This bug is meant to track all of the other bugs about making the MLKit camera samples perform better, so that it can be discussed in one place.

**Comment by samtstern** [comment]
Camera preview FPS very low: https://github.com/firebase/quickstart-android/issues/841

**Comment by samtstern** [comment]
OutOfMemoryError in ML Kit samples: https://github.com/firebase/quickstart-android/issues/775

**Comment by samtstern** [comment]
Low camera preview quality: https://github.com/firebase/quickstart-android/issues/832

**Comment by samtstern** [comment]
Update sample to Camera2 API: https://github.com/firebase/quickstart-android/issues/677

**Comment by samtstern** [comment]
Image processing is super battery hungry: https://github.com/firebase/quickstart-android/issues/703

**Comment by nhattv18** [comment]
> Camera preview FPS very low: #841

Hi samtstern,
We will discuss the issue here?

**Comment by samtstern** [comment]
@nhattv18 yep!  A lot of these issues are basically the same, we need to re-write most of the camera code in these samples to improve performance and use a more modern API.

**Comment by dev-gloomyfox** [comment]
OOM issue caused by convertBitmapToByteBuffer method.

When this method call, Direct ByteBuffer allocate every time.  When this ByteBuffer moved local variable, oom is not occurs.

**Comment by MrGilberts** [comment]
Where the normal people could see the progress. I was working in a solution but, it only works good in galaxy s9. the galaxy s7 its too slow , And i would like to see the progress.

**Comment by samtstern** [comment]
Our plan is to move directly from Camera1 to the new `camerax` library as soon as it gets out of `alpha` and into a `beta` (or higher) release.  This should improve the sample and hopefully remove a lot of code.

**Comment by bembem1011** [comment]
When does your team finish upgrading to cameraX to improve performance ? @samtstern

**Comment by samtstern** [comment]
We're working on it but the APIs for CameraX are still in alpha so we need
some more stability (and feature completeness) before we can release it.

**Comment by KaidoNinja** [comment]
Would really love to see this example with CameraX, any progress?

**Comment by aantelov87** [comment]
any progress?

**Comment by KaidoNinja** [comment]
CameraX is in Beta, so it's time to upgrade. :) 
https://developer.android.com/jetpack/androidx/releases/camera

**Comment by Anasmirza20** [comment]
Could you please let me know how much time it will take to upgrade the project using CameraX API.
